### PR TITLE
MAILPOET-4522 fix problems with marketing optin block and do watch js

### PR DESCRIPTION
**Pull request by samnajian**
Fixes [MAILPOET-4522]

[MAILPOET-4522]: https://mailpoet.atlassian.net/browse/MAILPOET-4522?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/webpack.config.js [hunk 1]
```diff
@@ -562,10 +554,9 @@ const marketingOptinBlock = Object.assign({}, wpScriptConfig, {
     extensions: ['.js', '.jsx', '.ts', '.tsx'],
   },
   plugins: [
-    ...wpScriptConfig.plugins.filter(
-      (plugin) =>
-        plugin.constructor.name !== 'DependencyExtractionWebpackPlugin',
-    ),
+    ...wpScriptConfig.plugins.filter((plugin) => {
+      return plugin.constructor.pluginName === 'mini-css-extract-plugin';
+    }),
```
Comment (costasovo) [inline mailpoet/webpack.config.js]: Could you please elaborate a bit on why this change was needed? Maybe even a comment in the code explaining why we filter the plugins like that could be useful.

File: mailpoet/webpack.config.js [hunk 2]
```diff
@@ -518,24 +518,16 @@ function requestToHandle(request) {
...
-      'assets/js/src/marketing_optin_block',
-      'frontend.ts',
-    ),
+    'marketing-optin-block': '/assets/js/src/marketing_optin_block/index.tsx',
+    'marketing-optin-block-frontend':
+      '/assets/js/src/marketing_optin_block/frontend.ts',
   },
   output: {
     filename: '[name].js',
-    path: path.resolve(process.cwd(), 'assets/dist/js/marketing_optin_block'),
```
Comment (costasovo) [inline mailpoet/webpack.config.js:538]: I just want to make sure I understand the change. The `process.cwd()` was causing the nesting, correct?

**Review by costasovo — COMMENTED** [review]
Good job finding the issue! I have just a couple of questions to better understand the fix.

File: mailpoet/webpack.config.js (hunk 2 above)
Comment (samnajian) [reply mailpoet/webpack.config.js:538]: Actually it was not this causing the nesting, but was looking weird compared to the rest of the config since we are always using `__dirname` in other parts of the config.

File: mailpoet/webpack.config.js (hunk 1 above)
Comment (samnajian) [reply mailpoet/webpack.config.js]: `CopyWebpackPlugin` was the one causing the problems and with the previous config we were allowing that to be used from `wpScriptConfig` even though we were adding our custom one as well.

**Review by costasovo — APPROVED** [review]
Thanks for explaining!

**Review by veljkho — APPROVED** [review]

**Comment by JanJakes** [comment]
@veljkho @samnajian This was created against the `js-gutenberg-update` branch and thus also merged into it. What now? Should we keep it there? I guess with PRs starting off of non-`trunk` branches we should wait requesting review before the parent is merged (and then rebase, to be sure).

**Comment by samnajian** [comment]
@JanJakes while what you're saying is correct, I guess for this case we can also directly merge these changes into trunk, what do you think? should I create a PR against trunk since this is already merged into this [js-gutenberg-update](https://github.com/mailpoet/mailpoet/tree/js-gutenberg-update)?

**Comment by JanJakes** [comment]
@samnajian Yep, I'd give priority to QAing the https://github.com/mailpoet/mailpoet/pull/4281, if possible, @veljkho 🙏

**Comment by veljkho** [comment]
#4281 will be my next task today.
